streams/src/main/java/org/apache/kafka/streams/state/internals/InMemoryTimeOrderedKeyValueBuffer.java

```diff
@@ -74,13 +77,17 @@
 
     private int partition;
 
-    public static class Builder implements StoreBuilder<StateStore> {
+    public static class Builder<K, V> implements StoreBuilder<StateStore> {
 
         private final String storeName;
+        private final Serde<K> keySerde;
+        private final Serde<V> valSerde;
         private boolean loggingEnabled = true;
 
-        public Builder(final String storeName) {
+        public Builder(final String storeName, final Serde<K> keySerde, final Serde<V> valSerde) {
```

Comment:
Just clarifying: are all the non-testing callers passing in the serdes, hence it would not be null in practice?

Reply:
hmm. Yes, although I think there's a problem with the logic. 

The KTableImpl does:
```java
            new InMemoryTimeOrderedKeyValueBuffer.Builder<>(storeName, keySerde, new FullChangeSerde<>(valSerde))
```

but `valSerde` might be null, and `FullChangeSerde` forbids null for this argument. So I think the default serde case doesn't work properly here.

I'll add a test for this and fix the code if I'm right.

Reply:
Ok, now the answer is "no". The only non-test caller is `KTableImpl`, and it will set its own serdes, but those might be `null` if they are not known.